File: Python/_1877_Minimize_Maximum_Pair_Sum_in_Array/Solution.py
# coding=utf-8
# Date: 2021/7/20 9:44
from typing import List


# 执行用时：304 ms, 在所有 Python3 提交中击败了74.83%的用户
# 内存消耗：25.6 MB, 在所有 Python3 提交中击败了40.95%的用户
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        i, j = 0, len(nums) - 1
        res = 0
        while i < j:
            res = max(res, nums[i] + nums[j])
            i += 1
            j -= 1
        return res


if __name__ == "__main__":
    print(Solution().minPairSum([3, 5, 2, 3]))  # 7

File: Java/src/_1877_Minimize_Maximum_Pair_Sum_in_Array/Solution.java
package _1877_Minimize_Maximum_Pair_Sum_in_Array;

import java.util.Arrays;

/**
 * Date: 2021/7/20 9:40
 * Content:
 * 执行用时：61 ms, 在所有 Java 提交中击败了95.60%的用户
 * 内存消耗：54 MB, 在所有 Java 提交中击败了57.42%的用户
 */
public class Solution {

    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int i = 0, j = nums.length - 1;
        int res = 0;
        while (i < j) res = Math.max(res, nums[i++] + nums[j--]);
        return res;
    }

    public static void main(String[] args) {

        int[] nums = {3, 5, 2, 3};
        System.out.println((new Solution()).minPairSum(nums));  // 7
    }
}
